pkg/epp/scheduling/filter.go

```diff
@@ -201,18 +211,20 @@ func lowLoRACostPredicate(req *LLMRequest, pod backendmetrics.PodMetrics) bool {
 // Returns:
 //   - Filtered slice of pod metrics based on affinity and availability
 //   - Error if any issues occur during filtering
-func loRASoftAffinityFilter(logger logr.Logger, req *LLMRequest, pods []backendmetrics.PodMetrics) ([]backendmetrics.PodMetrics, error) {
+func loRASoftAffinityFilterFunc(ctx *Context, pods []*PodMetrics) ([]*PodMetrics, error) {
 
 	// Pre-allocate slices with estimated capacity
-	filtered_affinity := make([]backendmetrics.PodMetrics, 0, len(pods))
-	filtered_available := make([]backendmetrics.PodMetrics, 0, len(pods))
+	filtered_affinity := make([]*PodMetrics, 0, len(pods))
+	filtered_available := make([]*PodMetrics, 0, len(pods))
 
 	// Categorize pods based on affinity and availability
 	for _, pod := range pods {
+		_, active := pod.ActiveModels[ctx.req.ResolvedTargetModel]
+		_, waiting := pod.WaitingModels[ctx.req.ResolvedTargetModel]
 
-		if _, exists := pod.GetMetrics().ActiveModels[req.ResolvedTargetModel]; exists {
+		if active || waiting {
```

Comment:
This is where we consider both active and waiting as lora affinity. This refactor just makes it super clear.